06-unittest.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: page
+title: Testing
+subtitle: The Unittest Module
+minutes: 10
+---
+> ## Learning Objectives {.objectives}
+>
+> -   Understand how to use Unittest to write and run test cases
+> -   Understand that Unittest can be used for many styles of test, not just unit tests
+> -   
+> -   
+
+The Python standard library provides a module for writing and running tests, called `unittest`, which is documented for [Python3.6](https://docs.python.org/3.6/library/unittest.html?highlight=unittest#module-unittest) and [Python2.7](https://docs.python.org/2/library/unittest.html?highlight=unittest#module-unittest).
+
+This module is useful for writing all styles of test, including unit tests and functional tests.
+
+## Functional Tests
+
+Functional testing is a way of checking software to ensure that it has all the required functionality that's specified within its functional requirements.  This will test many methods and may interact with dependencies like file access.
+
+Functional tests are often the first tests that are written as part of writing a program.  They can link closely to a requirement: "This programme shall ...".
+
+These tests are testing overall behaviour.
+
+They are especially useful if code is being restructured or re-factored, to build confidence that the key behaviour is maintained during the restructuring.
```

Comment:
behavior (since the rest of the lesson is american english, rather than british, we should probably keep it consistent.)